lib/authCodeGrant.js

```diff
@@ -182,14 +215,47 @@ function saveAuthCode (done) {
 }
 
 /**
+ * Save access token with model
+ *
+ * @param  {Function} done
+ * @this   OAuth
+ */
+function saveAccessToken (done) {
+  var accessToken = this.accessToken;
+
+  // Object idicates a reissue
+  if (typeof accessToken === 'object' && accessToken.accessToken) {
+    this.accessToken = accessToken.accessToken;
+    return done();
+  }
+
+  var expires = null;
+  if (this.config.accessTokenLifetime !== null) {
+    expires = new Date();
+    expires.setSeconds(expires.getSeconds() + this.config.accessTokenLifetime);
+  }
+
+  this.model.saveAccessToken(accessToken, this.client.clientId, expires,
+      this.user, function (err) {
+    if (err) return done(error('server_error', false, err));
+    done();
+  });
+}
+
+/**
  * Check client against model
  *
  * @param  {Function} done
  * @this   OAuth
  */
 function redirect (done) {
-  this.res.redirect(this.client.redirectUri + '?code=' + this.authCode +
-      (this.req.query.state ? '&state=' + this.req.query.state : ''));
+  if(this.responseType === 'code') {
+    this.res.redirect(this.client.redirectUri + '?code=' + this.authCode +
+        (this.req.query.state ? '&state=' + this.req.query.state : ''));
+  } else {
+    this.res.redirect(this.client.redirectUri + '#token=' + this.accessToken +
+        (this.req.query.state ? '#token=' + this.req.query.state : ''));
```

Comment:
I don't. The #token is important since I want to pass it on to a client. (Not the server)
I implemented it while looking at this article:
https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2#grant-type-implicit